File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file187.f90
subroutine print_array(d, ni, nl)
implicit none

DATA_TYPE, dimension(nl, ni) :: d
integer :: nl, ni
integer :: i, j
do i = 1, ni
do j = 1, nl
write(0, DATA_PRINTF_MODIFIER) d(j,i)

if (mod(((i - 1) * ni) + j - 1, 20) == 0) then
write(0, *)
end if

end do
end do
write(0, *)
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file187.cpp
#include <iostream>

template <typename DATA_TYPE>
void print_array(DATA_TYPE** d, int ni, int nl) {
    for (int i = 0; i < ni; ++i) {
        for (int j = 0; j < nl; ++j) {
            std::cout << d[j][i];

            if (((i * ni) + j) % 20 == 0) {
                std::cout << std::endl;
            }
        }
    }
    std::cout << std::endl;
}
